tests/test_workflow.py

```diff
@@ -43,3 +63,33 @@ def test_from_url(self, url: str) -> None:
         assert wf.name == "test"
         assert wf.kind == "Workflow"
         assert len(wf.spec.templates) == 1
+
+    def test_submit(self, api: V1alpha1Api, wf: Workflow) -> None:
+        """Test `Workflow.submit` method."""
+        fake_workflow_name = "test"
+        flexmock(V1alpha1Api).should_receive("create_namespaced_workflow").and_return(
+            fake_workflow_name
+        )
+
+        # submit w/o parameters
+        workflow_name: str = wf.submit(client=V1alpha1Api(), namespace="test")
+
+        assert isinstance(workflow_name, str)
+        assert workflow_name == "test"
+
+        # submit w/ parameters
+        with pytest.raises(AttributeError) as exc:
+            # Expected ValueError due to undefined parameter
+            workflow_name: str = wf.submit(
+                client=api, namespace="test", parameters={"param": "test"}
+            )
+
+        wf.spec.arguments = V1alpha1Arguments(
+            parameters=[V1alpha1Parameter(name="param")]
+        )
+        workflow_name: str = wf.submit(
```

Comment:
how can I pass parameters on submit for defined tasks? 
for eg I have a task like this
```
@task
@parameter(name="date", value='')
@parameter(name="userId", value='')
def fetchUserActivity(self, date: V1alpha1Parameter, userId: V1alpha1Parameter) -> V1alpha1Template:
    return self.someTemplateFunc(date, userId)
```

Reply:
These are task parameters, they are different from the `Workflow` level parameters and have nothing to do with the `submit` method.

Passing of the parameters internally in the worfklow, i.e. from `arguments` to `inputs`, is something that has to be improved. Currently, one has to use the full string according to the Workflow spec.

@brtsavpatel You can create an issue for it, if you wish.